OSVR-Unity/Assets/OSVRUnity/src/VRViewer.cs

```diff
@@ -117,7 +158,92 @@ public void UpdateEyes()
+                    _camera.enabled = true;
+                }
+                else
+                {
+                    // To save Render time, disable this camera here and re-enable after the frame
+                    // OR, in DirectMode, leave it on for "mirror" mode, although this is an expensive operation
+                    // The long-term solution is to provide a DirectMode preview window in RenderManager
+                    //@todo enable directmode preview in RenderManager
+                    _camera.enabled = DisplayController.UseRenderManager && DisplayController.showDirectModePreview;
+                }
+
+                DoRendering();
+
+                // Flag that we disabled the camera
+                _disabledCamera = true;
+            }
+
+            // The main rendering loop, should be called late in the pipeline, i.e. from OnPreCull
+            // Set our viewer and eye poses and render to each surface.
+            void DoRendering()
+            {
+                // update poses once DisplayConfig is ready
+                if (DisplayController.CheckDisplayStartup())
+                {
+                    // update the viewer's head pose
+                    // @todo Get viewer pose from RenderManager if UseRenderManager = true
+                    // currently getting viewer pose from DisplayConfig always
+                    UpdateViewerHeadPose(GetViewerPose(ViewerIndex));
+
+                    // each viewer updates its eye poses, viewports, projection matrices
+                    UpdateEyes();
+
+                }
+                else
+                {
+                    if (!DisplayController.CheckDisplayStartup())
+                    {
+                        //@todo do something other than not show anything
+                        Debug.LogError("Display Startup failed. Check HMD connection.");
+                    }
+                }
+            }
+
+            // This couroutine is called every frame.
+            IEnumerator EndOfFrame()
+            {
+                while (true)
+                {
+                    //if we disabled the dummy camera, enable it here
+                    if (_disabledCamera)
+                    {
+                        Camera.enabled = true;
+                        _disabledCamera = false;
+                    }
+                    yield return new WaitForEndOfFrame();
+                    if (DisplayController.UseRenderManager && DisplayController.CheckDisplayStartup())
+                    {
+                        // Issue a RenderEvent, which copies Unity RenderTextures to RenderManager buffers
+#if UNITY_5_2 || UNITY_5_3
+                        GL.IssuePluginEvent(DisplayController.RenderManager.GetRenderEventFunction(), OsvrRenderManager.RENDER_EVENT);
+#else
+                        Debug.LogError("GL.IssuePluginEvent failed. This version of Unity is not supported by RenderManager.");
```

Comment:
can we just issue this error once and then flip the UseRenderManager flag to false so they don't flood it